prelude/slang-cpp-scalar-intrinsics.h

```diff
@@ -126,7 +129,13 @@ SLANG_FORCE_INLINE int32_t I32_max(int32_t a, int32_t b) { return a > b ? a : b;
 SLANG_FORCE_INLINE int32_t I32_clamp(int32_t x, int32_t min, int32_t max) { return ( x < min) ? min : ((x > max) ? max : x); }
 
 SLANG_FORCE_INLINE float I32_asfloat(int32_t x) { Union32 u; u.i = x; return u.f; }
-SLANG_FORCE_INLINE uint32_t I32_asuint(int32_t x) { return uint32_t(x); } 
+SLANG_FORCE_INLINE uint32_t I32_asuint(int32_t x) { return uint32_t(x); }
+SLANG_FORCE_INLINE double I32_asdouble(int32_t low, int32_t hi )
+{
+    Union64 u;
+    u.i = (int64_t(hi) << 32) | low;
+    return u.d;
```

Comment:
Good catch on asdouble int coercion - yes that looks problematic.

On the 'union' pun, yes it isn't the spec strictly for C++, but is for C. On the other hand it is the 'de facto' way to do it, such that there is no modern compiler that doesn't support the idiom. It's hard to see a scenario where that will change - because doing so would break a huge amount of code, and what would be the advantage of doing so? All else being equal I believe it is generally better than the memcpy style.